Lib/test/test_httpservers.py

```diff
@@ -539,6 +539,35 @@ def test_get(self):
             finally:
                 os.chmod(self.tempdir, 0o755)
 
+    def test_range_get(self):
+        response = self.request(self.base_url + '/test')
+        self.assertEqual(response.getheader('accept-ranges'), 'bytes')
+        self.check_status_and_reason(response, HTTPStatus.OK, data=self.data)
+
+        response = self.request(self.base_url + '/test', headers={'Range': 'bytes=3-12'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 3-12/30')
+        self.assertEqual(response.getheader('content-length'), '10')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[3:13])
+
+        response = self.request(self.base_url + '/test', headers={'Range': 'bytes=3-'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 3-29/30')
+        self.assertEqual(response.getheader('content-length'), '27')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[3:])
+
+        response = self.request(self.base_url + '/test', headers={'Range': 'bytes=-5'})
+        self.assertEqual(response.getheader('content-range'), 'bytes 25-29/30')
+        self.assertEqual(response.getheader('content-length'), '5')
+        self.check_status_and_reason(response, HTTPStatus.PARTIAL_CONTENT, data=self.data[25:])
+
+        response = self.request(self.base_url + '/test', headers={'Range': 'bytes=100-200'})
```

Comment:
Can we have more tests for syntactically correct ranges but with invalid values?

Reply:
A few more tests have been added 😃That should cover all the code for now.